packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -280,11 +282,23 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         this._attachState = this.containerRuntime.attachState !== AttachState.Detached && this.existing ?
             this.containerRuntime.attachState : AttachState.Detached;
 
+        /**
+         * The existing flag can be true in two conditions:
+         * 1. It's a local data store that is created when a detached container is rehydrated. In this case, the data
+         *    store is locally visible because the snapshot it is loaded from contains locally visible data stores only.
+         * 2. It's a remote data store that is created when an attached container is loaded is loaded from snapshot or
+         *    when an attach op comes in. In both these cases, the data store is already globally visible.
+         */
+        if (existing) {
+            this.visibilityState = this.containerRuntime.attachState === AttachState.Detached
+                ? VisibilityState.LocallyVisible : VisibilityState.GloballyVisible;
+        }
+
         this.bindToContext = () => {
             assert(this.bindState === BindState.NotBound, 0x13b /* "datastore context is already in bound state" */);
             this.bindState = BindState.Binding;
             assert(this.channel !== undefined, 0x13c /* "undefined channel on datastore context" */);
-            bindChannelFn(this.channel);
+            makeVisibleFn();
```

Comment:
Some though on naming - should it be called makeVisibleLocallyFn()? It may go further, but I think that's what it actually does (everything else is fall out of that).

Reply:
And code below is very in line with that:
```ts
        this.visibilityState = VisibilityState.LocallyVisible;
        this.makeVisibleFn();
```

Reply:
The idea is that data store is already locally visible by this time and telling its parent that I am ready to be fully visible now. The parent decides whether it should become globally visible or not. The sequence of steps are:
1. Data store is created and is not visible.
2. It becomes locally visible when some one call `makeVisibleAndAttachGraph` (for root data stores) or `attachGraph` (for non-root data stores) on it.
3. The data store now tells its parent that I am locally visible so you can make me globally visible when you think the time is right.

So, maybe this function should be called `readyToBeVisible` or `readyToBeGloballyVisible`.